src/species_advance/species_advance_aos.h

```diff
@@ -135,6 +165,80 @@ class species_t {
 
             return global_id;
         }
+
+#ifdef VPIC_PARTICLE_ANNOTATION
+        /**
+         * @brief This allocates memory for user defined annotations on each particle
+         *
+         * @param slot_count how many float annotations per particle
+         *
+         * @ return No return value, dies if the allocation fails
+         */
+        void allocate_annotation_buffer(const int slot_count) {
+            if(slot_count <= 0) {
+                this->has_annotation = 0;
+                return;
+            }
+            this->has_annotation = slot_count;
+            MALLOC_ALIGNED(this->p_annotation, this->max_np * this->has_annotation * sizeof(float), 128);
+        }
+        /**
+         * @brief Retrieve as float value from an annotation slot
+         *
+         * @param particle_index index of particle 0 <= i < np
+         * @param slot_index of annotation within that particle 0 <= slot_index < has_annotation
+         *
+         * @return the stored annotation value
+         */
+        float get_annotation(const int particle_index, const int slot_index) {
+            if((particle_index < 0) || (particle_index >= this->np)) {
+                // Invalid particle index
+                return NAN;
```

Comment:
This scares me, but I'm not sure I have a better suggestion?

Reply:
I guess we could kill the run at that point.